gbpservice/nfp/orchestrator/modules/device_orchestrator.py

```diff
@@ -561,42 +619,83 @@ def create_network_function_device(self, event):
+            nfp_context['network_function_device'] = network_function_device
+
             # Create an event to NSO, to give device_id
             device_created_data = {
                 'network_function_instance_id': (
                     nfd_request['network_function_instance']['id']),
-                'network_function_device_id': device['id']
+                'network_function_device_id': device['id'],
             }
-            self._create_event(event_id='DEVICE_CREATED',
-                               event_data=device_created_data)
+            
             self._create_event(event_id='DEVICE_SPAWNING',
-                               event_data=device,
+                               event_data=nfp_context,
                                is_poll_event=True,
                                original_event=event)
 
-    @poll_event_desc(event='DEVICE_SPAWNING', spacing=20)
+            self._create_event(event_id='DEVICE_CREATED',
+                               event_data=device_created_data)
+
+    @poll_event_desc(event='DEVICE_SPAWNING', spacing=2)
     def check_device_is_up(self, event):
-        device = event.data
+        nfp_context = event.data
+
+        service_details= nfp_context['service_details']
+        network_function_device = nfp_context['network_function_device']
+        token = nfp_context['resource_owner_context']['auth_token']
+        tenant_id = nfp_context['resource_owner_context']['tenant_id']
+
+        device = {
+            'token': token,
+            'tenant_id': tenant_id,
+            'id': network_function_device['id'],
+            'service_details': service_details}
 
         orchestration_driver = self._get_orchestration_driver(
-            device['service_details']['service_vendor'])
+            service_details['service_vendor'])
+
         is_device_up = (
             orchestration_driver.get_network_function_device_status(device))
+
         if is_device_up == nfp_constants.ACTIVE:
+            # [(mak)TODO] - Update interfaces count here before
+            # sending health monitor rpc in DEVICE_UP event.
+            # [HACK] to handle a very corner case where 
+            # PLUG_INTERFACES completes later than HEALTHMONITOR.
+            # till proper fix is identified.
+            provider = nfp_context['provider']['ptg']
+            consumer = nfp_context['consumer']['ptg']
+            network_function_device = nfp_context['network_function_device']
+        
+            if provider:
+                network_function_device['interfaces_in_use'] += 1
+            if consumer:
+                network_function_device['interfaces_in_use'] += 1
+
             # create event DEVICE_UP
             self._create_event(event_id='DEVICE_UP',
-                               event_data=device,
+                               event_data=nfp_context)
+
+            self._create_event(event_id='PLUG_INTERFACES',
```

Comment:
yes we know this, currently this does not happen, as plug interfaces takes harldy 4-5secs to complete, while healthmonitor takes more time. But, yes i have a solution which can be implemented without synchronizing this events as it will increase latency.